lms/product/plugin/grouping.py

```diff
@@ -74,18 +74,27 @@ def sections_enabled(self, request, application_instance, course) -> bool:
         """Check if sections are enabled for this LMS, instance and course."""
         return bool(self.sections_type)
 
-    def get_group_set_id(self, request, assignment):
+    def get_group_set_id(self, _request, assignment, historical_assignment=None):
         """
         Get the group set ID for group launches.
 
         A course can be divided in multiple "small groups" but it's possible to
         have different sets of groups for the same course.
 
         This ID identifies a collection of groups.
+
+        `historical_assignment` might be none even when it might exist on the DB
+        (or even available as assignment.copied_form).
+        The historical_assignment is only relevant to get the config when creating an assignment for the first time.
         """
-        if not self.group_type or not assignment:
+        if not self.group_type:
+            # Groups not enabled on this product
             return None
 
+        if not assignment and historical_assignment:
```

Comment:
I've been staring at this and I'm not sure I quite grok it. Is it this?

```python
        if assignment:
            return assignment.extra.get("group_set_id")

        if historical_assignment:
            # When creating new assignments, take the value from the previous
            # version of the assignment
            return historical_assignment.extra.get("group_set_id")

        return None
```

Because I think that's the same logic as in the `misc.py:get_document_url`.